### DISCUSSION TITLE
Support Larger Palette

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#16322] [Feature requests]
The fixed palette in rct2 is "good enough" but what if we increased it. Maybe 16 bit color? How would that be implemented?

**Comment by duncanspumpkin** [comment]
I don't really see this happening until we have a complete new graphics set it's certainly possible to do just look at openttd.

**Comment by 733737** [comment]
openttd is "open" source, could the code be copied into openrct2 with permission?

**Comment by ocalhoun6** [comment]
There shouldn't be any _legal_ issues with copying openttd code ... but it's really unlikely to be a 1:1 fit for OpenRCT2 code, so you can't just copy and paste and call it done. _Maybe_ you'd be able to adapt snippets of openttd code to work within this codebase, but that's probably not going to be much better than simply doing it from scratch.

**Comment by 733737** [comment]
it is using the same engine, but I know it's a different iteration

**Reply by Lastorder-DC** [reply]
It's **not** using same engine. Made by same dev does not mean code is shared between games.

**Reply by 733737** [reply]
sorry I assumed it was since it appears similar with isometric sprites.

**Reply by ZehMatt** [reply]
CS games usually do have a lot in common code wise so thats not a far fetched assumption. However OpenTTD has a way different code style and data structures so this won't really help.

**Reply by pizza2004** [reply]
Yeah the problem is that while it _might_ be accurate to describe the original RCT as being built on a newer version of the TTD engine, they were reverse engineered by two different groups using different methods and with different ideologies, so the end result is that they aren't really cross compatible in most ways.  I think early on OpenRCT did grab some of the UI code from OpenTTD, or something like that, as a way to speed up the progress on some of the most similar bits of code, but I wasn't directly involved with that so I could be wrong and it was also 6 years ago and both codebases have changed significantly since then.
